test/extended/apiserver/api_requests.go

```diff
@@ -71,16 +77,100 @@ var _ = g.Describe("[sig-arch][Late]", func() {
+					"customresourcedefinitions.v1beta1.apiextensions.k8s.io",
+					"rolebindings.v1beta1.rbac.authorization.k8s.io",
+					"roles.v1beta1.rbac.authorization.k8s.io",
+				),
+				bugzilla: "https://bugzilla.redhat.com/show_bug.cgi?id=1947797",
+			},
+
+			"system:serviceaccount:openshift-machine-api:cluster-autoscaler-operator": {
+				resourcesUsed: sets.NewString(
+					"validatingwebhookconfigurations.v1beta1.admissionregistration.k8s.io",
+				),
+				bugzilla: "https://bugzilla.redhat.com/show_bug.cgi?id=1947785",
+			},
+
+			"system:serviceaccount:openshift-machine-config-operator:default": {
+				resourcesUsed: sets.NewString(
+					"customresourcedefinitions.v1beta1.apiextensions.k8s.io",
+					"certificatesigningrequests.v1beta1.certificates.k8s.i",
+				),
+				bugzilla: "https://bugzilla.redhat.com/show_bug.cgi?id=1947791",
+			},
+
+			"system:serviceaccount:openshift-operator-lifecycle-manager:olm-operator-serviceaccount": {
+				resourcesUsed: sets.NewString(
+					"customresourcedefinitions.v1beta1.apiextensions.k8s.io",
+					"certificatesigningrequests.v1beta1.certificates.k8s.io",
+				),
+				bugzilla: "https://bugzilla.redhat.com/show_bug.cgi?id=1947794",
+			},
+
+			"system:admin": {
+				resourcesUsed: sets.NewString(
+					"certificatesigningrequests.v1beta1.certificates.k8s.io",
+					"clusterrolebindings.v1beta1.rbac.authorization.k8s.io",
+					"customresourcedefinitions.v1beta1.apiextensions.k8s.io",
+					"ingresses.v1beta1.extensions",
+					"rolebindings.v1beta1.rbac.authorization.k8s.io",
+					"roles.v1beta1.rbac.authorization.k8s.io",
+				),
+				bugzilla: "current an exception while we add user-agent",
+			},
+		}
+
+		unexpectedFailures := []string{}
 		for user, resourceToRequestCount := range userToResourceToRequestCount {
+			// ignore usage by our e2e test for now.
+			if strings.HasPrefix(user, "e2e-test-") {
+				continue
+			}
 			for resource, requestCount := range resourceToRequestCount {
 				details := fmt.Sprintf("user/%v accessed %v %d times", user, resource, requestCount.count)
 				failureOutput = append(failureOutput, details)
 				framework.Logf(details)
+
+				if !buggedUserToResource[user].resourcesUsed.Has(resource) {
+					unexpectedFailures = append(unexpectedFailures, details)
+				}
 			}
 		}
 
-		sort.Strings(failureOutput)
+		unusedSkips := []string{}
+		for user, resourceException := range buggedUserToResource {
+			for _, resource := range resourceException.resourcesUsed.List() {
+				if _, ok := userToResourceToRequestCount[user][resource]; !ok {
+					details := fmt.Sprintf("zz -- unused exception for user/%v and resource %v", user, resource)
+					unusedSkips = append(unusedSkips, details)
+					framework.Logf(details)
+				}
+			}
+		}
 
+		sort.Strings(unexpectedFailures)
+		if len(unexpectedFailures) > 0 {
+			g.Fail(strings.Join(unexpectedFailures, "\n"))
+			return
+		}
+
+		sort.Strings(unusedSkips)
+		if len(unusedSkips) > 8 {
```

Comment:
where does `8` come from?

Reply:
> where does `8` come from?

you need some slack in order to allow things to improve.  But you also need motivation to clean out entries that are no longer required.  I chose eight as big enough to make progress and small enough to keep the list tidy.